File: payment-service/src/main/java/com/example/paymentservice/service/PaymentService.java
package com.example.paymentservice.service;

import com.example.paymentservice.dto.request.CreatePaymentRequest;
import com.example.paymentservice.dto.response.PageResponse;
import com.example.paymentservice.dto.response.PaymentResponse;
import com.example.paymentservice.dto.response.StripeCheckoutResponse;

public interface PaymentService {
    PaymentResponse createPayment(String userId, String token, CreatePaymentRequest request);

    PageResponse<PaymentResponse> getMyPayments(String userId, int page, int size);

    PageResponse<PaymentResponse> getAllPayments(int page, int size);

    PaymentResponse getPaymentDetail(String userId, String paymentId);

    PaymentResponse markPaymentSuccess(String userId, String token, String paymentId);

    PaymentResponse markPaymentFailed(String userId, String paymentId);

    PaymentResponse cancelPayment(String userId, String paymentId);

    StripeCheckoutResponse createStripeCheckout(String userId, String paymentId);

    PaymentResponse reconcileStripePayment(String userId, String paymentId);

    void handleStripeWebhook(String payload, String signature);
}

File: web-app/src/services/paymentService.js
import httpClient from "../configurations/httpClient";
import { API } from "../configurations/configuration";

const defaultPage = (params = {}) => ({
  content: [],
  currentPage: 1,
  pageSize: params.size || 10,
  totalPages: 0,
  totalElements: 0,
});

export const createPayment = async (request) => {
  const response = await httpClient.post(API.PAYMENTS, request);
  return response.data?.data;
};

export const createStripeCheckout = async (paymentId) => {
  const response = await httpClient.post(
    `${API.PAYMENTS}/${paymentId}/stripe-checkout`,
  );
  return response.data?.data;
};

export const reconcileStripePayment = async (paymentId) => {
  const response = await httpClient.post(
    `${API.PAYMENTS}/${paymentId}/stripe-reconcile`,
  );
  return response.data?.data;
};

export const getMyPayments = async (params = {}) => {
  const response = await httpClient.get(`${API.PAYMENTS}/my-payments`, {
    params,
  });

  return response.data?.data || defaultPage(params);
};

export const getAllPayments = async (params = {}) => {
  const response = await httpClient.get(API.PAYMENTS, {
    params,
  });

  return response.data?.data || defaultPage(params);
};

export const getPaymentDetail = async (id) => {
  const response = await httpClient.get(`${API.PAYMENTS}/${id}`);
  return response.data?.data;
};

export const markPaymentSuccess = async (id) => {
  const response = await httpClient.put(`${API.PAYMENTS}/${id}/success`);
  return response.data?.data;
};

export const markPaymentFailed = async (id) => {
  const response = await httpClient.put(`${API.PAYMENTS}/${id}/failed`);
  return response.data?.data;
};

export const cancelPayment = async (id) => {
  const response = await httpClient.put(`${API.PAYMENTS}/${id}/cancel`);
  return response.data?.data;
};
